### PR TITLE
Update OpenTelemetry.Instrumentation.Owin to depend on at least Owin 4.2.2

### LINKED ISSUE #386: Add support to the current version of the Microsoft.Owin
# Issue with OpenTelemetry.Instrumentation.Owin

`OwinTelemetry.Instrumentation.Owin` is linked to the exact `4.1.1` version of the `Microsoft.Owin` package which was updated on `9/8/2020`.
https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/main/build/Common.props#L30
```xml
<MicrosoftOwinPkgVer>[4.1.1]</MicrosoftOwinPkgVer>
```
This prevents the updating to the newer version of the package.
The current latest version of the package is [4.2.2](https://www.nuget.org/packages/Microsoft.Owin/).

List of [all OpenTelemetry NuGet packages](https://www.nuget.org/profiles/OpenTelemetry) and version that you are
using (e.g. `OpenTelemetry 1.0.2`):

- `OpenTelemetry.Instrumentation.Owin 1.0.0-rc.2`

Runtime version (e.g. `net462`, `net48`, `netcoreapp3.1`, `net6.0` etc. You can
find this information from the `*.csproj` file):

`net48`

**Is this a feature request or a bug?**

* [x] Feature Request
* [ ] Bug

**What is the expected behavior?**

The package should support the latest version of the `Microsoft.Owin` package.

What do you expect to see?

- Version range for the library is changed to:
```xml
<MicrosoftOwinPkgVer>[4.1.1; 5.0)</MicrosoftOwinPkgVer>
```

**What is the actual behavior?**

- The update to the current version of the `Microsoft.Owin` package is not possible.

What did you see instead? If you are reporting a bug, create a self-contained
project using the template of your choice and apply the minimum required code to
result in the issue you're observing. We will close this issue if:

-

## Additional Context

Add any other context about the feature request here.

### PR DESCRIPTION
**Pull request by najadojo**
Fixes #386

## Changes

Changed to depend on at least Owin 4.2.2 to resolve a [denial of service vulnerability](https://github.com/advisories/GHSA-3rq8-h3gj-r5c6).

For significant contributions please make sure you have completed the following items:

* [X] Appropriate `CHANGELOG.md` updated for non-trivial changes
* [ ] Design discussion issue #

### CODE REVIEW

File: build/Common.props [hunk 1]
```diff
@@ -28,7 +28,7 @@
     <MicrosoftExtensionsHostingAbstractionsPkgVer>[2.1.0,5.0)</MicrosoftExtensionsHostingAbstractionsPkgVer>
     <MicrosoftExtensionsOptionsPkgVer>[3.1.0,)</MicrosoftExtensionsOptionsPkgVer>
     <MicrosoftNETFrameworkReferenceAssembliesPkgVer>[1.0.0,2.0)</MicrosoftNETFrameworkReferenceAssembliesPkgVer>
-    <MicrosoftOwinPkgVer>[4.1.1]</MicrosoftOwinPkgVer>
+    <MicrosoftOwinPkgVer>[4.2.2,)</MicrosoftOwinPkgVer>
```
Comment (utpilla) [inline build/Common.props:31]: Keep this as `[4.1.1,5.0)`? The instrumentation package doesn't really need `4.2.2`. So, if someone wanted to use it with `4.1.1` they still can. Also, 5.0+ could possibly bring in breaking changes so let's wait till it's released and see if the instrumentation library needs any changes.

Comment (najadojo) [reply build/Common.props:31]: <4.2.2 has a [vulnerability](https://github.com/advisories/GHSA-3rq8-h3gj-r5c6) so I'd rather not propagate usage of that version of the package.

Comment (CodeBlanch) [reply build/Common.props:31]: @utpilla It looks like a [security thing](https://github.com/advisories/GHSA-3rq8-h3gj-r5c6) so I'm good with bumping to `4.2.2`. Regarding if we should have the upper bound, I'm good with that if that's what you want to do!

**Review by CodeBlanch — APPROVED** [review]
LGTM

File: src/OpenTelemetry.Instrumentation.Owin/CHANGELOG.md [hunk 2]
```diff
@@ -2,9 +2,12 @@
 
 ## Unreleased
 
+## 1.0.0-rc.3
+
 * Changed activity source name from `OpenTelemetry.OWIN`
   to `OpenTelemetry.Instrumentation.Owin`
   ([#572](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/572))
+* Changed to depend on at least Owin 4.2.2 to resolve a [denial of service vulnerability](https://github.com/advisories/GHSA-3rq8-h3gj-r5c6).
```
Comment (utpilla) [inline src/OpenTelemetry.Instrumentation.Owin/CHANGELOG.md:10]: This has to be placed in the unreleased section. You should also add the PR link.

File: build/Common.props (hunk 1 above)
Comment (utpilla) [reply build/Common.props]: Okay. That's good! I wasn't aware of it.

**Review by utpilla — APPROVED** [review]
Approved with suggestions. Please add the link to the PR in a follow-up PR. You could refer to the previous CHANGELOG entry to see the format in which PR links are added: 
`([#648](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/648))`

**Pull request merged**